AWSAuthSDK/Tests/AWSMobileClientTests/AWSMobileClientHostedUITests.swift

```diff
@@ -0,0 +1,68 @@
+//
+//  AWSMobileClientHostedUITests.swift
+//  AWSMobileClientHostedUITests
+//
+
+import XCTest
+
+@testable import AWSMobileClient
+import AWSAuthCore
+import AWSCognitoIdentityProvider
+
+class AWSMobileClientHostedUITests: AWSMobileClientTestBase {
+    
+    static var scopes = ["openid", "email"]
+    
+    override class func setUp() {
+        // signs out the user
+        super.setUp()
+        XCTAssertNil(AWSMobileClient.default().scopes)
+    }
+    
+    static func setHostedUIScopesInKeychain() {
+        AWSMobileClient.default().scopes = scopes
+        AWSMobileClient.default().saveHostedUIOptionsScopesInKeychain()
+        
+        // assert if custom scopes are correctly stored in the keychain
+        AWSMobileClient.default().loadHostedUIScopesFromKeychain()
+        XCTAssertEqual(AWSMobileClient.default().scopes, scopes)
+    }
+    
+    func reInitializeMobileClient() {
```

Comment:
Why its called reInitialize? AWSMobileClient can be initialized only once.

Reply:
We are testing if on app restart(mobile client default instance is reintialized) would fetch the custom scopes if they exist in keychain

Reply:
I dont think we can test it properly in unit test. AWSMobileClient doesnot support reinitialization.